src/main/java/uk/gov/ons/ctp/response/action/service/ActionPlanJobService.java

```diff
@@ -60,88 +71,71 @@ public ActionPlanJob findActionPlanJob(final UUID actionPlanJobId) {
     return actionPlanJobRepo.findByActionPlanFK(actionPlan.getActionPlanPK());
   }
 
-  public List<ActionPlanJob> createAndExecuteAllActionPlanJobs() {
-    final List<ActionPlanJob> executedJobs = new ArrayList<>();
-    actionPlanRepo
-        .findAll()
-        .forEach(
-            actionPlan -> {
-              final Date lastExecutionTime =
-                  new Date(
-                      nowUTC().getTime() - appConfig.getPlanExecution().getDelayMilliSeconds());
-              if (actionPlan.getLastRunDateTime() == null
-                  || actionPlan.getLastRunDateTime().before(lastExecutionTime)) {
-                ActionPlanJob job =
-                    ActionPlanJob.builder()
-                        .actionPlanFK(actionPlan.getActionPlanPK())
-                        .createdBy(CREATED_BY_SYSTEM)
-                        .build();
-                job = createAndExecuteActionPlanJob(job);
-                if (job != null) {
-                  executedJobs.add(job);
-                }
-              } else {
-                log.debug(
-                    "Job for plan {} has been run since last wake up - skipping",
-                    actionPlan.getActionPlanPK());
-              }
-            });
-    return executedJobs;
+  public void createAndExecuteAllActionPlanJobs() {
+    List<ActionPlan> actionPlans = actionPlanRepo.findAll();
+    actionPlans.forEach(
+        actionPlan -> {
+          if (hasActionPlanBeenRunSinceLastSchedule(actionPlan)) {
```

Comment:
I would break this if... else out into a separate function, because it's quite a lot of logic to have buried in a lambda